web/src/components/graph/areagraph.js

```diff
@@ -0,0 +1,132 @@
+import React, {
+  useState,
+  useEffect,
+  useRef,
+  useMemo,
+} from 'react';
+import { scaleTime, scaleLinear } from 'd3-scale';
+import { first, last } from 'lodash';
+import moment from 'moment';
+
+import AreaGraphGradients from './areagraphgradients';
+import AreaGraphLayers from './areagraphlayers';
+import GraphBackground from './graphbackground';
+import GraphHoverLine from './graphhoverline';
+import ValueAxis from './valueaxis';
+import TimeAxis from './timeaxis';
+
+const X_AXIS_HEIGHT = 20;
+const Y_AXIS_WIDTH = 35;
+const Y_AXIS_PADDING = 4;
+
+const getTimeScale = (containerWidth, datetimes, currentTime) => scaleTime()
+  .domain([first(datetimes), currentTime ? moment(currentTime).toDate() : last(datetimes)])
+  .range([0, containerWidth]);
+
+const getValueScale = (containerHeight, maxTotalValue) => scaleLinear()
+  .domain([0, maxTotalValue * 1.1])
+  .range([containerHeight, Y_AXIS_PADDING]);
+
+const AreaGraph = React.memo(({
+  id,
```

Comment:
why is this needed?

Reply:
There was this chunk in the styles:

```scss
#country-history-mix {
    height: 10rem;

    .layer path {
        opacity: 1.0

        &:hover {
            opacity: 0.8;
        }
    }
}
```

However it seems that the opacity effects weren't really working and the height is something that can probably be fixed for now (please let me know if you think there would be value in passing it as a prop instead) - updated in https://github.com/tmrowco/electricitymap-contrib/pull/2171/commits/23c405160f733db0116c0583e52ff3c9a32247b8.